pyproject.toml

```diff
@@ -39,7 +39,7 @@ dependencies = [
   "multidict>=6.0.2",
   "polyfactory>=2.6.3",
   "pyyaml",
-  "typing-extensions",
+  "typing-extensions>=4.6.0",
```

Comment:
Why do we need this constraint?

Reply:
I propose to leave this alone, because typing-exntensions is a common source of resolution problems. **Unless** we relly on some details in the implementation.

Reply:
We use `Buffer` in an isinstance check which was added in 4.6. If the use of the constraint outweighs the utility of using the type I'll remove its use. Let me know what you think.

Reply:
This check is only being done in the `pydantic_init_plugin` right? If that's the case, I think we can safely remove the constraint since `pydantic` sets the minimum constraint for `typing_extensions` to `4.6.1`. Or we could change that to `isinstance(value, (bytes, bytearray, memoryview))`.

Reply:
> sets the minimum constraint for typing_extensions to 4.6.1

This is the clincher for me, we'll just remove the constraint on our side :heavy_check_mark: